jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile

```diff
@@ -17,15 +17,31 @@ pipeline {
 
         stage('Clone Operator Repo') {
             steps {
-                //ToDo Clone the integeratly-operator repo
-                echo "Clone Operator Repo"
+                cleanWs()
+                git 'https://github.com/integr8ly/integreatly-operator'
             }
         }
 
         stage("Ensure Product Branch") {
             steps {
-                //ToDo Create the product next branch i.e. 3scale-next from the target branch if it doesn't exist
-                echo "Ensure Product Branch"
+                withCredentials([usernamePassword(credentialsId: 'githubjenkins', passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')]) {
+                    sh("""
+                        git config --global user.name 'Automatron'
+                        git config --global user.email 'github.eng@feedhenry.com'
+                        git config --global credential.username ${GIT_USERNAME}
+                        git config --global credential.helper "!echo password=${GIT_PASSWORD}; echo"
+                       """)
+                }
+                script {
+                    def checkout = sh(script: 'git checkout ${PRODUCT_BRANCH}', returnStatus: true)
+                    if (checkout) {
+                        sh("""
+                            git checkout -B ${PRODUCT_BRANCH}
+                            git push origin ${PRODUCT_BRANCH}
+                           """)
+                    }
+                    sh 'git rebase master'
```

Comment:
This needs to be configurable. We should be able to set the target branch somewhere and have it be used everywhere.

Reply:
I'll add it as a parameter to the job.